### DISCUSSION TITLE
MicrosoftDB for PostgreSQL (Citus)

### DISCUSSION THREAD

**Discussion by JaivyDaam** [brwilkinson/AzureDeploymentFramework#18] [Ideas]
Hi, after looking at your awesome work, and I am still getting into it, I've seen that there is no Citus Bicep file.
I'll do my outmost best to integrate this, as I really like the work you've put in here. I see that with my own bicep files it is hard to keep consistency across the resources naming and the way template inherit one-another.

Thank you for this framework, I'm looking forward into using it :-)

**Comment by brwilkinson** [comment]
Oh nice. glad to hear. I do use certain patterns, so if you are able to pick up on the patterns, then hopefully you can find it is easy to extend and modify to your needs.

If you get stuck on anything let me know.

I am just in the process of moving to bicepparam files... as you will see in the "AD" tenant.

- https://github.com/brwilkinson/AzureDeploymentFramework/tree/main/ADF/tenants/AD

I am not sure if you are able to actually deploy with my script with the tenant model or if you are just using the templates for samples?

- It's a bit of setting up to get things working with my Global and Regional dependencies before deploying a stamp.

e.g. 
```powershell
AzSet -App AD -Enviro P0
AzDeploy @Current -Prefix ACU1 -TF ADF:\bicep\PostgreSQL.bicep` etc.
```

Anyway if you need guidance, let me know...

Also i you have any requests for certain resource providers I am happy to add.

If I ever get questions on the Bicep repo, I nearly always add the provider type into this project to test it... then other providers I actually use for my production purposes, so it just depends how rich certain templates are, if I was actually using that provider in my production.

**Reply by JaivyDaam** [reply]
Thank you for your extensive answer and your willingness to help me out. Unfortunately, I am having troubles getting this to run locally on my Mac as a lot of the Powershell modules are not compatible or plainly not working (Add-AppxPackage for example).

I am not big on Powershell but I am Azure focussed (I just don't like GCP and AWS), so the `ADF/1-prereqs/00-InstallTools.ps1` is not working for me out of the box. As I understood, Powershell should be Powershell, but seems like it isn't :-). 

I guess the best way to go about it, is to fork the repo, make the changes in the configurations and use Azure Cloud Shell? Or alternatively, get a Windows PC/Laptop/VM?

I do have the azure-cli with the extensions needed to deploy Bicep files with parameters. I am just trying to figure out the deployment hierarchy as I am working from 1 subscription (at the moment, that will change over the next 6 months) and as such management groups and subscription creations are not of interest to me, yet.

For my solution that I am trying to use the ADF for, needs the Hub&Spoke configurations. Where it is important that Peering is enable to the Hub but not between Spokes, where each region (Continental or otherwise) will get their own Hub.

I already have some resources that are being actively used like DNS Zones, Static Web Apps, Container Apps, App Services, Keyvaults, IoT Hub and others, but it has become a real mess. 
I would love to migrate it to using your patterns and naming conventions because they make much more sense and are way more consistent then anything I have.

As there are live users using it, I am rather 'afraid' and would like to be "as sure as possible" that the deployment I do will not mess that up :-)

Just so you know what my ADF onboarding looks like:
- Fork the Repository
- Get a Windows machine
- Install the requirements
- Create Tenant
  - This part kind of confuses me a tiny bit as I am unsure how I would define a tenant in my current setup but I guess this is something for me to figure out
- Make the changes as stated on the [Getting Started](https://brwilkinson.github.io/AzureDeploymentFramework/Getting_Started.html) page
- Deploy Orchestration/Base template

If I am missing anything or you have tips&tricks please do let me know :-)

**Reply by brwilkinson** [reply]
Okay, thank you for the feedback.

This should run fine on a Mac with powershell 7.

That other setup script is quite old and was just written for Windows, so that is windows specific.

All of the actual pre-reqs should be cross platform. E.g. Vscode, powershell 7, the AZ modules. 

Then some other command line tools, oh-my-posh and posh-git. 

Normally I do setup with my customers, that I work with on projects. So it's hard to get this up and running on your own. I might make videos once Bicep reached version 1.0.1. However until then we expect some breaking changes, so I haven't created any actual "release" that would have to be stable. 

Perhaps I can get access to a Mac somehow and try to create a setup doc specifically for Mac. 

So you know what a devcontainer is? This project is setup to run in a devcontainer, so I think that would be the easiest way to get started. You can do it locally or on github.
